tools/walletextension/main/main.go

```diff
@@ -3,25 +3,47 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"net"
+	"time"
 
 	"github.com/obscuronet/go-obscuro/tools/walletextension"
 )
 
 const (
+	tcp       = "tcp"
 	localhost = "127.0.0.1"
 )
 
 func main() {
 	config := parseCLIArgs()
+
+	// We wait thirty seconds for a connection to the node. If we cannot establish one, we exit the program.
+	fmt.Printf("Waiting up to thirty seconds for connection to host at %s...\n", config.NodeRPCHTTPAddress)
+	counter := 30
+	for {
```

Comment:
This hides other issues now e.g. if the port is in use it will just spin for 30 seconds and then exit as below;

```
Waiting up to thirty seconds for connection to host at 127.0.0.1:13000...
Could not establish connection to host at 127.0.0.1:13000. Exiting
```

... with no indication of what the problem would be. If the port was in use previously you would have seen the error as below;

```
Morays-MacBook-Pro:main moraygrieve$ ./wallet_extension 
Wallet extension config: 
{
  "WalletExtensionPort": 3000,
  "NodeRPCHTTPAddress": "127.0.0.1:13000",
  "NodeRPCWebsocketAddress": "127.0.0.1:13001",
  "LogPath": "wallet_extension_logs.txt"
}
Wallet extension started.
💡 Visit http://127.0.0.1:3000/viewingkeys/ to generate an ephemeral viewing key.
panic: listen tcp 127.0.0.1:3000: bind: address already in use
```

Reply:
Those are two separate errors. One is the host being unavailable, the other is the local port you want to bind to serve up the wallet extension being unavailable. I guess my change means that you don't detect the latter error, but only because you're detecting the former. Are you saying we should add an extra check here to see if the local port is also free?

Reply:
Just that it hides the error response in the implementation ... i.e. before you would have seen the port was in use and so would fix the issue quickly, whereas now that information is lost to the user and they don't immediately realise the underlying cause. Yes you can't establish connection to the server, but the issue is not on the server side but on your side.